src/virtualenv/seed/wheels/periodic_update.py

```diff
@@ -220,16 +220,17 @@ def trigger_update(distribution, for_py_version, wheel, search_dirs, app_data, e
     kwargs = {"stdout": pipe, "stderr": pipe}
     if not debug and sys.platform == "win32":
         kwargs["creationflags"] = CREATE_NO_WINDOW
-    process = Popen(cmd, **kwargs)  # noqa: S603
-    logging.info(
-        "triggered periodic upgrade of %s%s (for python %s) via background process having PID %d",
-        distribution,
-        "" if wheel is None else f"=={wheel.version}",
-        for_py_version,
-        process.pid,
-    )
-    if debug:
-        process.communicate()  # on purpose not called to make it a background process
+
+    with Popen(cmd, **kwargs) as process:  # noqa: S603
```

Comment:
What happens here? Wouldn't this make it blocking call?

Reply:
Yes, that's the point. You can't keep a subprocess open while your main process exits without getting a `ResourceWarning`, afaik. Is the intention to keep this process running until it terminates even if the main process finishes?

If you want to continue the rest of the program execution and only wait for the subprocess to finish at the end you'd need to register the process somewhere central and ensure termination.

Another option maybe is to use [`close_fds` param of `Popen`](https://docs.python.org/3/library/subprocess.html#popen-constructor), but I'm not sure if that fixes the issue.

Reply:
Ideally, I'd like to just ignore the resource warning. There's no need for this process to finish when the main process finishes. It's more of a fire and forget kinda thing.

Reply:
Nevermind, it doesn't. I was running with `--no-periodic-update`.

Reply:
Yes sorry, after a bunch of research: the proper way to do this is to wait until your subprocess finishes before your application exits. You don't want a zombie process hanging around. Even solutions using asyncio recommend to wait for async tasks to finish before main process exit, else you'll get a warning.

Reply:
> You don't want a zombie process hanging around.

That's exactly what I want here though.

Reply:
So, the unclosed file warning stems from setting `pipe = PIPE` in the non-debug case. It disappears if we set it to `pipe=subprocess.DEVNULL` instead. 

But then we get a different `ResourceWarning`: `subprocess is still running`

The warning was added in 3.6, here: https://bugs.python.org/issue26741
Pretty much everyone saying keeping a zombie process is bad practice, but here we go.

The followup issue mentions some potential solutions to hide the warning, but `detach` didn't work: https://bugs.python.org/issue27068

So I had a look at the CPython source, turns out if we set the returncode manually (since we don't care about it anyway), the warning will not be raised: https://github.com/python/cpython/blob/85e5b1f5b806289744ef9a5a13dabfb23044f713/Lib/subprocess.py#L1127 (from main, just to make it a permalink)

Tests passed locally.